cognee/infrastructure/databases/graph/neo4j_driver/adapter.py

```diff
@@ -141,21 +142,29 @@ async def has_node(self, node_id: str) -> bool:
         )
         return results[0]["node_exists"] if len(results) > 0 else False
 
-    async def add_node(self, node: DataPoint):
+    async def add_node(
+        self, node: Union[DataPoint, str], properties: Optional[Dict[str, Any]] = None
+    ) -> None:
         """
-        Add a new node to the database based on the provided DataPoint object.
+        Add a new node to the database based on the provided DataPoint object or string ID.
 
         Parameters:
         -----------
 
-            - node (DataPoint): An instance of DataPoint representing the node to add.
-
-        Returns:
-        --------
-
-            The result of the query execution, typically the ID of the added node.
+            - node (Union[DataPoint, str]): An instance of DataPoint or string ID representing the node to add.
+            - properties (Optional[Dict[str, Any]]): Properties to set on the node when node is a string ID.
         """
-        serialized_properties = self.serialize_properties(node.model_dump())
+        if isinstance(node, str):
+            # TODO: this was not handled in the original code, check if it is correct
+            # Handle string node ID with properties parameter
+            node_id = node
```

Comment:
note to self. Shall we deprecate to `node: Union[DataPoint]` since we don't support `str` anyways?